Add LBP6300/LBP6300dn PPD to captdriver.

Generates a PPD that CUPS can present to the user as "Canon LBP6300/LBP6300dn"
with the rastertocapt filter and appropriate resolution/media options.

diff --git a/src/canon-lbp.drv b/src/canon-lbp.drv
index 1111111..2222222 100644
--- a/src/canon-lbp.drv
+++ b/src/canon-lbp.drv
@@ -1,6 +1,6 @@
 /*
  * Canon CAPT LBP Printers
- * captdriver PPD Source File
+ * captdriver PPD Source File - patched with LBP6300 support
  * for use with CUPS ppdc to generate PPDs
  *
  */
@@ -276,4 +276,141 @@
 		PCFileName "CNLB3K10.ppd"
 	}
 
+	{
+		// LBP6300, LBP6300dn
+
+                // Darkness
+                Group "PrinterSettings/Printer Settings"
+                Option "captDarkness/Darkness" PickOne AnySetup 10
+		Choice 0 "<</cupsInteger1    0>>setpagedevice"
+		Choice 1 "<</cupsInteger1    1>>setpagedevice"
+		Choice 2 "<</cupsInteger1    2>>setpagedevice"
+		Choice 3 "<</cupsInteger1    3>>setpagedevice"
+		Choice 4 "<</cupsInteger1    4>>setpagedevice"
+		Choice 5 "<</cupsInteger1    5>>setpagedevice"
+		Choice 6 "<</cupsInteger1    6>>setpagedevice"
+		*Choice 7 "<</cupsInteger1   7>>setpagedevice"
+		Choice 8 "<</cupsInteger1    8>>setpagedevice"
+		Choice 9 "<</cupsInteger1    9>>setpagedevice"
+		Choice 10 "<</cupsInteger1  10>>setpagedevice"
+		Choice 11 "<</cupsInteger1  11>>setpagedevice"
+		Choice 12 "<</cupsInteger1  12>>setpagedevice"
+		Choice 13 "<</cupsInteger1  13>>setpagedevice"
+		Choice 14 "<</cupsInteger1  14>>setpagedevice"
+		Choice 15 "<</cupsInteger1  15>>setpagedevice"
+
+                // Resolution
+                *Resolution k 1 160 595 0 "600x600dpi/600 DPI"
+                Resolution k 1 160 595 0 "600x400dpi/400 DPI"
+
+		/* LBP 6300/6300dn, with multi-page size and type support */
+		MediaType 1   "Plain/Plain Paper"
+		MediaType 4   "Heavy/Heavy Paper"
+		MediaType 8   "PlainL/Plain Paper L"
+		MediaType 5   "HeavyH/Heavy Paper H"
+		MediaType 24  "Transparency"
+                MediaType 4   "Label"
+		MediaType 20  "Envelope"
+
+                MinSize 75mm 105mm
+		MaxSize 215.9mm 355.6mm		// US Legal
+                HWmargins 4.7095835mm 4.7095835mm 4.709583mm 4.709583mm
+
+		/*
+		 * NOTE: Some of the sizes appear to be identical to
+		 * another at least in terms of width and length. There
+		 * could be other differences, and may be useful in managing
+		 * multiple trays on printers that support them.
+		 *
+		 */
+
+		/* Common Document Sizes */
+		VariablePaperSize yes
+		*MediaSize A4
+		MediaSize A5
+		MediaSize B5
+		MediaSize Executive	// US 7.25x10.5 inch
+		MediaSize Folio	// SE Asian F4
+		MediaSize Legal
+		MediaSize Letter
+		MediaSize SISG5
+		MediaSize SISE5
+		MediaSize SAC16K
+		MediaSize SAC32K
+
+		/* Less Common Document Sizes
+		 *
+		 * PROTIP: Delete this section and recompile PPD, if page size
+		 * menus in your most used apps clutter up and cannot be managed.
+		 * Any size below may be moved to the list above, before deleting
+		 * this section.
+		 */
+		MediaSize 8x10
+		MediaSize A4Small	// Same dimensions as A4
+		MediaSize A4Plus	// Longer than Folio (F4) by just one point
+		MediaSize A5Extra
+		MediaSize A6
+		MediaSize B6
+		MediaSize B7
+		MediaSize DoublePostcard
+		MediaSize FanFoldGerman
+		MediaSize FanFoldGermanLegal
+		MediaSize ISOB5
+		MediaSize ISOB5Extra
+		MediaSize ISOB6
+		MediaSize ISOB7
+		MediaSize LetterPlus
+		MediaSize Note		// Same dimensions as Letter (US)
+		MediaSize Postcard
+		MediaSize PRC16K
+		MediaSize PRC32K
+		MediaSize PRC32KBig
+		MediaSize Quarto
+		MediaSize Statement
+
+		/* Common Envelope Sizes */
+		HWmargins 10mm 10mm 10mm 10mm
+		MediaSize Env10
+		MediaSize EnvC5
+		MediaSize EnvChou4
+		MediaSize EnvDL
+		MediaSize EnvItalian
+
+		/* Less Common Envelope Sizes */
+		/*
+		 * PROTIP: Delete this section as necessary. If any size below
+		 * is needed, move it to the list above before deleting this section.
+		 *
+		 */
+		MediaSize 3x5
+		MediaSize 7x9
+		MediaSize C6
+		MediaSize Env9
+		MediaSize Env11
+		MediaSize Env12
+		MediaSize Env14
+		MediaSize EnvC6
+		MediaSize EnvC65
+		MediaSize EnvChou3
+		MediaSize EnvISOB5
+		MediaSize EnvISOB6
+		MediaSize EnvKaku3
+		MediaSize EnvMonarch
+		MediaSize EnvPersonal
+		MediaSize EnvPRC1
+		MediaSize EnvPRC2
+		MediaSize EnvPRC3
+		MediaSize EnvPRC4
+		MediaSize EnvPRC5
+		MediaSize EnvPRC6
+		MediaSize EnvPRC7
+		MediaSize EnvPRC8
+		MediaSize EnvYou4
+
+		// PPD file name
+		*ModelName "LBP6300/LBP6300dn r2c"
+		FileName "CanonLBP-6300-6300dn.ppd"
+		PCFileName "CNLB63DN.ppd"
+	}
+
 }
